src/main/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImpl.java

```diff
@@ -124,42 +124,60 @@ public CompanyData createCompanyData(final CompanySpec spec) throws DataExceptio
         final String companyNumberPrefix = spec.getJurisdiction().getCompanyNumberPrefix(spec);
 
         do {
-            // company number format: PP+123456 (Prefix either 0 or 2 chars, example uses 2 chars)
             spec.setCompanyNumber(companyNumberPrefix
                     + randomService
                     .getNumber(COMPANY_NUMBER_LENGTH - companyNumberPrefix.length()));
         } while (companyProfileService.companyExists(spec.getCompanyNumber()));
 
         try {
             companyProfileService.create(spec);
+            LOG.info("Successfully created company profile");
+
             filingHistoryService.create(spec);
+            LOG.info("Successfully created filing history");
+
             appointmentService.create(spec);
+            LOG.info("Successfully created appointments");
+
             var authCode = companyAuthCodeService.create(spec);
+            LOG.info("Successfully created auth code: " + authCode.getAuthCode());
+
             companyMetricsService.create(spec);
+            LOG.info("Successfully created company metrics");
+
             companyPscStatementService.create(spec);
+            LOG.info("Successfully created PSC statement");
+
             companyPscsService.create(spec);
+            LOG.info("Successfully created PSCs");
 
             if (spec.getRegisters() != null && !spec.getRegisters().isEmpty()) {
+                LOG.info("Creating company registers for company: " + spec.getCompanyNumber());
                 this.companyRegistersService.create(spec);
+                LOG.info("Successfully created company registers");
             }
 
             String companyUri = this.apiUrl + "/company/" + spec.getCompanyNumber();
-
             var companyData = new CompanyData(spec.getCompanyNumber(),
                     authCode.getAuthCode(), companyUri);
 
-            // Add company to the elastic search index
             if (isElasticSearchDeployed) {
+                LOG.info("Adding company to ElasticSearch index: " + spec.getCompanyNumber());
                 this.companySearchService.addCompanyIntoElasticSearchIndex(companyData);
+                LOG.info("Successfully added company to ElasticSearch index");
             }
+
+            LOG.info("Successfully created all company data for: " + spec.getCompanyNumber());
             return companyData;
         } catch (Exception ex) {
             Map<String, Object> data = new HashMap<>();
             data.put("company number", spec.getCompanyNumber());
-            LOG.error("Rolling back creation of company", data);
+            LOG.info("Failed to create company data for: " + spec.getCompanyNumber(), data);
```

Comment:
Could the caught exception message also be logged here? I'm aware it may be caught and logged later but this gives us a single log line to look for